logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/ConversationEventReceiver.kt

```diff
@@ -173,11 +172,16 @@ class ConversationEventReceiver(
                 if (isSenderVerified(message.content.messageId, message.conversationId, message.senderUserId))
                     messageRepository.markMessageAsDeleted(messageUuid = message.content.messageId, conversationId = message.conversationId)
                 else kaliumLogger.i(message = "Delete message sender is not verified: $message")
-            is MessageContent.DeleteForMe ->
+            is MessageContent.DeleteForMe -> {
                 if (message.conversationId == conversationRepository.getSelfConversationId())
-                //todo: consider to check with conversation id
-                    messageRepository.deleteMessage(messageUuid = message.content.messageId)
+                    messageRepository.deleteMessage(
+                        messageUuid = message.content.messageId,
+                        /*The conversationId comes with the hidden message[message.content] only carries the conversaionId VALUE,
+                        *  we need to get the DOMAIN from the self conversationId[here is the message.conversationId]*/
+                        conversationId = ConversationId(message.content.conversationId, message.conversationId.domain)
```

Comment:
⚠️ This may not work on federated backends.


Take the scenario:
- I am a member of Backend X (my self conversation ID is: `selfConvoValue@backendX`)
- Conversation A is in Backend Y (ID is `convA@backendY`)
- I hide a message on Conversation A

The domain from self conversation is different from the conversation I am hiding the message from.


This... sucks big time. And I am not sure we can improve it other than using the `value` only when hiding messages, and try to ignore the `domain`.

Reply:
Thanks @vitorhugods, good point! I'll take care of that.